### PR TITLE
set the error message when disconnected from the server

### PR DESCRIPTION
**Pull request by cu-kai**
This will open an error message to the client's main menu stating why they have been disconnected from the server (whether kicked or otherwise).

### CODE REVIEW

**Review by illwieckz — APPROVED** [review]
LGTM.

**Comment by cu-kai** [comment]
Note: the colours are not yet parsed by the `<inlinecvar>` tags in RML and there will be a companion PR by @illwieckz to fix that.

File: src/engine/client/cl_cgame.cpp [hunk 1]
```diff
@@ -149,8 +149,10 @@ bool CL_HandleServerCommand(Str::StringRef text, std::string& newText) {
 	if (cmd == "disconnect") {
 		// NERVE - SMF - allow server to indicate why they were disconnected
 		if (argc >= 2) {
+			Cvar_Set( "com_errorMessage", va( "^3Server disconnected:\n^7%s", args.Argv(1).c_str() ) );
```
Comment (DolceTriade) [inline src/engine/client/cl_cgame.cpp:152]: Use Cvar::Set() and Str::Format so you don't need the c_str().

**Comment by cu-kai** [comment]
I've now changed this to always provide an error message where `Sys::Drop` is called. This should also remove the need to search the console for an error when booted from the game in most cases.

**Review by DolceTriade — APPROVED** [review]

**Comment by slipher** [comment]
I found that ClientApplication::OnDrop is the perfect place to set it to get only important errors.

```diff
From 9895658466dea7514ffca2e90a5a578f60ffad1b Mon Sep 17 00:00:00 2001
From: slipher <slipher@protonmail.com>
Date: Sun, 15 Oct 2023 01:15:04 -0500
Subject: [PATCH] Set com_errorMessage only for non-shutdown errors

Avoid setting it for an error in e.g. cgame shutdown which is not that
important and could mask a more important error e.g. a ban message.

Also this avoids setting it in any other case where it is not useful -
non-client applications or fatal (for the engine) errors.
---
 src/common/System.cpp                   | 2 --
 src/engine/client/ClientApplication.cpp | 3 +++
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/common/System.cpp b/src/common/System.cpp
index 5dd8d999..bc636e25 100644
--- a/src/common/System.cpp
+++ b/src/common/System.cpp
@@ -178,8 +178,6 @@ void Drop(Str::StringRef message)
                errorCount = 0;
        lastError = now;

-       Cvar::SetValue( "com_errorMessage", Str::Format( "^3%s", message ) ); // always show an error to the user when dropping
-
        throw DropErr(true, message);
 }

diff --git a/src/engine/client/ClientApplication.cpp b/src/engine/client/ClientApplication.cpp
index fd87891b..6757b98d 100644
--- a/src/engine/client/ClientApplication.cpp
+++ b/src/engine/client/ClientApplication.cpp
@@ -95,6 +95,9 @@ class ClientApplication : public Application {
             SV_Shutdown(Str::Format("Server %s: %s", error ? "crashed" : "shutdown", reason).c_str());
             CL_Disconnect(true);
             CL_ShutdownAll();
+            if (error) {
+                Cvar::SetValue("com_errorMessage", Str::Format("^3%s", reason));
+            }
             CL_StartHunkUsers();
         }

--
2.29.2.windows.3
```

**Pull request merged**